update bioconductor packages

pabloaledo opened the pull request:
This PR updates several bioconductor packages to use more standard names, and also use commit-id's instead of version numbers for better reproducibility.

pabloaledo committed:
update bioconductor packages

marcodelapierre commented:
@alalazo  FYI

alalazo reviewed (dismissed):
Looks good to me. Let me check how to deal with the name changes. I assume these packages are widely used, so I want to double check if we need anything more than have an entry in the CHANGELOG for the v0.22 release.

marcodelapierre commented:
@pabloaledo this should have come to my mind earlier. See this old PR of mine where I wanted to rename e.g. `tower-cli` to `nf-tower-cli`: https://github.com/spack/spack/pull/34515/files

Instead of renaming, I was asked to duplicate the package recipes, and `deprecate` the old ones. Those will then be removed in a subsequent release of Spack.

@alalazo  I think this would address your concern, right?

alalazo commented:
@marcodelapierre Yes, that works for sure. Apologies for not getting back earlier, but tried to contact previous maintainer without much success.

marcodelapierre commented:
No worries, thank you Max!

pabloaledo committed:
keep previous versions of dupradar, rsubread and tximeta for compatibility

consider marking them deprecated since previous commit introduces better naming and versioning

Signed-off-by: Pablo <pablo.aledo@seqera.io>

marcodelapierre commented:
@pabloaledo  please add deprecation attributes as per: https://github.com/spack/spack/pull/34515/files#diff-6c3a337d07dc30fd0ede96af6f5cf4d84df89afc0886807aea5abea1f7819c59

on the actual removal: just noticed that my packages are still there one year later, opening a PR for removal now :-)

pabloaledo committed:
add deprecation attribute

Signed-off-by: Pablo <pablo.aledo@seqera.io>

marcodelapierre commented:
@alalazo  we have re-added the old named recipes (`bioconductor-` instead of `r-`), with the `deprecate` attribute. 

Once the latest `develop` is merged in (@pabloaledo ),  it hould be safe to merge into Develop!

alalazo commented on var/spack/repos/builtin/packages/r-ebseq/package.py:
```diff
@@ -6,7 +6,7 @@
 
 
-class BioconductorEbseq(RPackage):
+class REbseq(RPackage):
```
If i read the diff correctly, this one was not restored and deprecated.

pabloaledo replied:
Sorry!, forgot that one :facepalm:

alalazo replied:
No worries, thanks!

pabloaledo committed:
restore and deprecate bioconductor-ebseq

Signed-off-by: Pablo <pablo.aledo@seqera.io>

alalazo commented:
Thanks!

marcodelapierre commented:
Thank you Max!
